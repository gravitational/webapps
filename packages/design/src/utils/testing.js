import React from 'react';
import {
  render as testingRender,
  screen,
  fireEvent,
} from '@testing-library/react';
import ThemeProvider from 'design/ThemeProvider';
import theme from 'design/theme';
import 'jest-styled-components';

export function render(component) {
  return testingRender(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export { screen, fireEvent, theme };
