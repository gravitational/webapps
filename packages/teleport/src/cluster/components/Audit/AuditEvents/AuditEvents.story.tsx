/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import AuditEvents from './AuditEvents';
import { makeEvent } from 'teleport/services/audit';

export default {
  title: 'Teleport/Audit/Events',
};

export const Events = () => <AuditEvents {...defaultProps} />;

const events = [
  {
    code: 'T2004I',
    ei: 10,
    event: 'session.end',
    namespace: 'default',
    sid: '9febab45-6491-11e9-80a1-427cfde50f5a',
    time: '2019-04-22T00:00:51.543Z',
    uid: '6bf836ee-197c-453e-98e5-31511935f22a',
    user: 'admin@example.com',
  },
  {
    code: 'T1000I',
    event: 'user.login',
    method: 'local',
    success: true,
    time: '2019-04-22T00:49:03Z',
    uid: '173d6b6e-d613-44be-8ff6-f9f893791ef2',
    user: 'admin@example.com',
  },
  {
    code: 'T3007W',
    error:
      'ssh: principal "fsdfdsf" not in the set of valid principals for given certificate: ["root"]',
    event: 'auth',
    success: false,
    time: '2019-04-22T02:09:06Z',
    uid: '036659d6-fdf7-40a4-aa80-74d6ac73b9c0',
    user: 'admin@example.com',
  },
  {
    code: 'T1000W',
    error: 'user(name="fsdfsdf") not found',
    event: 'user.login',
    method: 'local',
    success: false,
    time: '2019-04-22T18:06:32Z',
    uid: '597bf08b-75b2-4dda-a578-e387c5ce9b76',
    user: 'fsdfsdf',
  },
  {
    'addr.local': '172.31.28.130:3022',
    'addr.remote': '151.181.228.114:51454',
    code: 'T2000I',
    ei: 0,
    event: 'session.start',
    login: 'root',
    namespace: 'default',
    server_id: 'de3800ea-69d9-4d72-a108-97e57f8eb393',
    sid: '56408539-6536-11e9-80a1-427cfde50f5a',
    size: '80:25',
    time: '2019-04-22T19:39:26.676Z',
    uid: '84c07a99-856c-419f-9de5-15560451a116',
    user: 'admin@example.com',
  },
  {
    code: 'T2002I',
    ei: 3,
    event: 'resize',
    login: 'root',
    namespace: 'default',
    sid: '56408539-6536-11e9-80a1-427cfde50f5a',
    size: '80:25',
    time: '2019-04-22T19:39:52.432Z',
    uid: '917d8108-3617-4273-ab37-7bbf8e7c1ab9',
    user: 'admin@example.com',
  },
  {
    'addr.local': '172.31.28.130:3022',
    'addr.remote': '151.181.228.114:51752',
    code: 'T2001I',
    ei: 4,
    event: 'session.join',
    login: 'root',
    namespace: 'default',
    server_id: 'de3800ea-69d9-4d72-a108-97e57f8eb393',
    sid: '56408539-6536-11e9-80a1-427cfde50f5a',
    time: '2019-04-22T19:39:52.434Z',
    uid: '13d26190-289b-41d4-af67-c8c8b0617ebe',
    user: 'admin@example.com',
  },
  {
    action: 'download',
    'addr.local': '172.31.28.130:3022',
    'addr.remote': '127.0.0.1:55594',
    code: 'T3004I',
    event: 'scp',
    login: 'root',
    namespace: 'default',
    path: '~/fsdfsdfsdfsdfs',
    time: '2019-04-22T19:41:23Z',
    uid: '183ca6de-c24b-4f67-854f-163c01245fa1',
    user: 'admin@example.com',
  },
];

const defaultProps = {
  attempt: {},
  attemptActions: {
    do: () => null,
  },
  onFetchLatest: () => Promise.resolve(),
  onFetch: () => Promise.resolve(),
  searchValue: '',
  events: events.map(e => makeEvent(e)),
};
