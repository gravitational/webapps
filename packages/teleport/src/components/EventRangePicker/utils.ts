import moment from 'moment';

export function getRangeOptions(): EventRange[] {
  return [
    {
      name: 'Today',
      from: moment(new Date())
        .startOf('day')
        .toDate(),
      to: moment(new Date())
        .endOf('day')
        .toDate(),
    },
    {
      name: '7 days',
      from: moment()
        .subtract(6, 'day')
        .startOf('day')
        .toDate(),
      to: moment(new Date())
        .endOf('day')
        .toDate(),
    },
    {
      name: 'Custom Range...',
      isCustom: true,
      from: new Date(),
      to: new Date(),
    },
  ];
}

export type EventRange = {
  from: Date;
  to: Date;
  isCustom?: boolean;
  name?: string;
};
