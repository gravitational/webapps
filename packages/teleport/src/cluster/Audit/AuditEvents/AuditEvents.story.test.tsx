import React from 'react';
import { Events } from './AuditEvents.story';
import { render } from 'design/utils/testing';

test('rendering of Events', () => {
  const { container } = render(<Events />);
  expect(container).toMatchSnapshot();
});
