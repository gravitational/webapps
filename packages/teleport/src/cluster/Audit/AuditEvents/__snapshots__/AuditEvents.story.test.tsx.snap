// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of Events 1`] = `
.c3 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 400;
  font-size: 12px;
  line-height: 16px;
  margin: 0px;
  color: #FFFFFF;
}

.c6 {
  display: inline-block;
  transition: color .3s;
  color: #FFFFFF;
  font-size: 16px;
}

.c8 {
  display: inline-block;
  transition: color .3s;
  color: #FFFFFF;
}

.c10 {
  display: inline-block;
  transition: color .3s;
  margin-right: 16px;
  padding: 4px;
  color: #FFFFFF;
  font-size: 16px;
}

.c13 {
  display: inline-block;
  transition: color .3s;
  margin-left: 8px;
  margin-right: -8px;
  color: rgba(255,255,255,0.56);
  font-size: 14px;
}

.c7 {
  background: #222C59;
  box-shadow: 0 4px 16px rgba(0,0,0,0.24);
  border-collapse: collapse;
  border-spacing: 0;
  border-radius: 8px;
  font-size: 12px;
  width: 100%;
}

.c7 > thead > tr > th,
.c7 > tbody > tr > th,
.c7 > tfoot > tr > th,
.c7 > thead > tr > td,
.c7 > tbody > tr > td,
.c7 > tfoot > tr > td {
  padding: 8px 8px;
  vertical-align: middle;
}

.c7 > thead > tr > th:first-child,
.c7 > tbody > tr > th:first-child,
.c7 > tfoot > tr > th:first-child,
.c7 > thead > tr > td:first-child,
.c7 > tbody > tr > td:first-child,
.c7 > tfoot > tr > td:first-child {
  padding-left: 24px;
}

.c7 > thead > tr > th:last-child,
.c7 > tbody > tr > th:last-child,
.c7 > tfoot > tr > th:last-child,
.c7 > thead > tr > td:last-child,
.c7 > tbody > tr > td:last-child,
.c7 > tfoot > tr > td:last-child {
  padding-right: 24px;
}

.c7 > thead > tr > th {
  background: #111B48;
  color: #FFFFFF;
  cursor: pointer;
  font-size: 10px;
  font-weight: 400;
  padding-bottom: 0;
  padding-top: 0;
  text-align: left;
  opacity: 0.75;
  text-transform: uppercase;
  white-space: nowrap;
}

.c7 > thead > tr > th .c5 {
  font-weight: bold;
  font-size: 8px;
  margin-left: 8px;
}

.c7 > tbody > tr > td {
  color: rgba(255,255,255,0.87);
  line-height: 16px;
}

.c7 tbody tr {
  border-bottom: 1px solid #1C254D;
}

.c7 tbody tr:hover {
  background-color: rgb(37,49,98);
}

.c12 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #2C3A73;
  border: 1px solid #1C254D;
  opacity: .87;
  color: rgba(255,255,255,0.87);
  font-size: 10px;
  min-height: 24px;
  padding: 0px 16px;
  height: 24px;
}

.c12:active {
  opacity: 0.56;
}

.c12:hover,
.c12:focus {
  background: #2C3A73;
  border: 1px solid rgba(255,255,255,0.1);
  opacity: 1;
}

.c12:active {
  opacity: 0.24;
}

.c12:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c14 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #2C3A73;
  border: 1px solid #1C254D;
  opacity: .87;
  color: rgba(255,255,255,0.87);
  font-size: 10px;
  min-height: 24px;
  padding: 0px 16px;
  width: 87px;
}

.c14:active {
  opacity: 0.56;
}

.c14:hover,
.c14:focus {
  background: #2C3A73;
  border: 1px solid rgba(255,255,255,0.1);
  opacity: 1;
}

.c14:active {
  opacity: 0.24;
}

.c14:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c0 {
  box-sizing: border-box;
  margin-bottom: 24px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.c4 {
  box-sizing: border-box;
  display: flex;
}

.c4 button {
  background: none;
  border: none;
  border-radius: 200px;
  cursor: pointer;
  padding: 0;
  margin: 0 0 0 8px;
  outline: none;
  transition: all 0.3s;
  text-align: center;
}

.c4 button:hover,
.c4 button:focus {
  background: #1C254D;
}

.c4 button:hover .c5,
.c4 button:focus .c5 {
  opacity: 1;
}

.c4 button .c5 {
  opacity: 0.4;
  font-size: 20px;
  transition: all 0.3s;
}

.c2 {
  padding: 8px 24px;
  display: flex;
  height: 24px;
  flex-shrink: 0;
  align-items: center;
  justify-content: space-between;
  background: #222C59;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
}

.c11 {
  background: #000;
  border: 2px solid #651FFF;
  color: rgba(255,255,255,0.87);
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 20px;
  height: 20px;
  padding: 0;
  border-radius: 100px;
  transition: all 0.3s;
}

.c11:hover,
.c11:active,
.c11:focus {
  box-shadow: 0 4px 16px rgba(0,0,0,0.24);
  color: #FFFFFF;
}

.c11:active {
  box-shadow: none;
  opacity: 0.56;
}

.c9 {
  display: flex;
  align-items: center;
  min-width: 130px;
  font-size: 12px;
  font-weight: 500;
  line-height: 24px;
  white-space: nowrap;
}

.c1 {
  box-sizing: border-box;
  font-size: 12px;
  min-width: 200px;
  outline: none;
  border: none;
  border-radius: 200px;
  height: 32px;
  transition: all .2s;
  background: #222C59;
  margin-right: 16px;
  padding-left: 16px;
  padding-right: 16px;
  color: rgba(255,255,255,0.87);
  height: 30px;
}

.c1:hover {
  background: #2C3A73;
}

.c1:focus,
.c1:active {
  background: #2C3A73;
  box-shadow: inset 0 2px 4px rgba(0,0,0,.24);
  color: rgba(255,255,255,0.87);
}

.c1::placeholder {
  color: rgba(255,255,255,0.24);
  font-size: 12px;
}

<div>
  <div
    class="sc-AxgMl c0"
  >
    <input
      class="c1"
      color="text.primary"
      height="30px"
      placeholder="SEARCH..."
      value=""
    />
  </div>
  <div>
    <nav
      class="c2"
    >
      <div
        class="c3"
        color="primary.contrastText"
      >
        SHOWING 
        <strong>
          1
        </strong>
         - 
        <strong>
          32
        </strong>
         
        of 
        <strong>
          32
        </strong>
      </div>
      <div
        class="sc-AxgMl c4"
      >
        <button
          disabled=""
          title="Previous Page"
        >
          <span
            class="c5 c6 icon icon-arrow-left-circle "
            color="light"
            font-size="3"
          />
        </button>
        <button
          disabled=""
          title="Next Page"
        >
          <span
            class="c5 c6 icon icon-arrow-right-circle "
            color="light"
            font-size="3"
          />
        </button>
      </div>
    </nav>
    <table
      class="c7"
    >
      <thead>
        <tr>
          <th>
            <a>
              Type
              <span
                class="c5 c8 icon icon-chevrons-expand-vertical "
                color="light"
              />
            </a>
          </th>
          <th>
            Description
          </th>
          <th>
            <a>
              Created
              <span
                class="c5 c8 icon icon-chevron-up "
                color="light"
              />
            </a>
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              class="c9"
            >
              <a
                href="/web/cluster/localhost/session/941a4c65-c6cd-11ea-9bef-482ae3513733"
                style="text-decoration: none;"
                target="_blank"
                title="Open Session Player"
              >
                <span
                  class="c5 c10 icon icon-terminal c11"
                  color="light"
                  font-size="3"
                />
              </a>
              Session Ended
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [root] has ended an interactive session lasting 2 minutes [941a4c65-c6cd-11ea-9bef-482ae3513733] on node [im-a-nodename]
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-07-15 19:03:05
          </td>
          <td
            align="right"
          >
            <button
              class="c12"
              height="24px"
              kind="border"
            >
              OPTIONS
              <span
                class="c5 c13 icon icon-caret-down "
                color="text.secondary"
                font-size="2"
              />
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              OIDC Auth Connector Created
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [unimplemented] created OIDC connector [new_oidc_connector]
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 19:29:14
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              GITHUB Auth Connector Deleted
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [unimplemented] deleted Github connector [new_github_connector]
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 19:28:28
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              GITHUB Auth Connector Created
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [unimplemented] created Github connector [new_github_connector] has been created
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 19:28:00
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              User Password Updated
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [Ivan_Jordan] has changed a password
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 19:26:53
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Access Request Updated
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Access request [66b827b2-1b0b-512b-965d-6c789388d3c9] has been updated to APPROVED
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 19:26:53
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Access Request Created
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Access request [66b827b2-1b0b-512b-965d-6c789388d3c9] has been created and is PENDING
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 19:26:53
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Reset Password Token Created
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [b331fb6c-85f9-4cb0-b308-3452420bf81e.one] created a password reset token for user [hello]
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 16:24:22
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              User Created
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [hello] has been created
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 16:24:05
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              User Updated
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [bob] has been updated
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 16:24:05
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              User Deleted
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [bob] has been deleted
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 16:24:05
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Data
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Usage report has been updated for session [5fc8bf85-a73e-11ea-afd1-0242ac0a0101]
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 15:14:51
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Data
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Usage report has been updated for session [5fc8bf85-a73e-11ea-afd1-0242ac0a0101]
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-06-05 15:14:51
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Command
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Program [ping] has been executed within a session [44c6cea8-362f-11ea-83aa-125400432324]
          </td>
          <td
            style="min-width: 120px;"
          >
            2020-01-13 18:05:53
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-get_app c5 c10"
                color="light"
                font-size="3"
              />
              SCP Download
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] downloaded a file [~/fsdfsdfsdfsdfs] from node [172.31.28.130:3022]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:41:23
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-get_app c5 c10"
                color="light"
                font-size="3"
              />
              SCP Download
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] downloaded a file [~/fsdfsdfsdfsdfs] from node [172.31.28.130:3022]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:41:23
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              User Joined
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] has joined the session [56408539-6536-11e9-80a1-427cfde50f5a]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:52
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              User Joined
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] has joined the session [56408539-6536-11e9-80a1-427cfde50f5a]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:52
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Terminal Resize
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] resized the session [56408539-6536-11e9-80a1-427cfde50f5a] terminal
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:52
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Terminal Resize
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] resized the session [56408539-6536-11e9-80a1-427cfde50f5a] terminal
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:52
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Started
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] has started a session [56408539-6536-11e9-80a1-427cfde50f5a]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:26
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Started
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] has started a session [56408539-6536-11e9-80a1-427cfde50f5a]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:26
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session File Access
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Program [bash] accessed a file [/etc/profile.d/] within a session [44c6cea8-362f-11ea-83aa-125400432324]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:26
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Network Connection
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Program [bash] opened a connection [10.217.136.161 &lt;-&gt; 190.58.129.4:3000] within a session [44c6cea8-362f-11ea-83aa-125400432324]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 19:39:26
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Local Login Failed
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Local user [fsdfsdf] login failed [user(name="fsdfsdf") not found]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 18:06:32
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Local Login Failed
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Local user [fsdfsdf] login failed [user(name="fsdfsdf") not found]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 18:06:32
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Auth Attempt Failed
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] failed auth attempt: ssh: principal "fsdfdsf" not in the set of valid principals for given certificate: ["root"]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 02:09:06
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Auth Attempt Failed
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] failed auth attempt: ssh: principal "fsdfdsf" not in the set of valid principals for given certificate: ["root"]
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 02:09:06
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Local Login
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Local user [admin@example.com] successfully logged in
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 00:49:03
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-info_outline c5 c10"
                color="light"
                font-size="3"
              />
              Local Login
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            Local user [admin@example.com] successfully logged in
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 00:49:03
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Ended
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] has ended a non-interactive session [9febab45-6491-11e9-80a1-427cfde50f5a] on node [node-hostname] 
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 00:00:51
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              class="c9"
            >
              <span
                class="c5 c10 icon icon-terminal c5 c10"
                color="light"
                font-size="3"
              />
              Session Ended
            </div>
          </td>
          <td
            style="word-break: break-word;"
          >
            User [admin@example.com] has ended a non-interactive session [9febab45-6491-11e9-80a1-427cfde50f5a] on node [undefined] 
          </td>
          <td
            style="min-width: 120px;"
          >
            2019-04-22 00:00:51
          </td>
          <td
            align="right"
          >
            <button
              class="c14"
              kind="border"
              width="87px"
            >
              Details
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
